pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -1026,6 +1043,127 @@ func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.Contr
 	return &registriesIgn, nil
 }
 
+func (ctrl *Controller) createDefaultContainerRuntimeMC(cfg *mcfgv1.ContainerRuntimeConfig) error {
+	// Check if the crio-default-container-runtime config map exists in the openshift-machine-config-operator namespace
+	defaultContainerRuntimeCM, err := ctrl.kubeClient.CoreV1().ConfigMaps(ctrl.namespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error checking for %s config map: %w", configMapName, err)
+	}
+	// If the crio-default-container-runtime config map exists, the MC was already created, so skip creating it again.
+	if defaultContainerRuntimeCM != nil && !errors.IsNotFound(err) {
+		return nil
+	}
+
+	sel, err := metav1.LabelSelectorAsSelector(metav1.AddLabelToSelector(&metav1.LabelSelector{}, builtInLabelKey, ""))
+	if err != nil {
+		return err
+	}
+	// Find all the MachineConfigPools
+	mcpPoolsAll, err := ctrl.mcpLister.List(sel)
+	if err != nil {
+		return err
+	}
+
+	// Create the crio-default-container-runtime MC for all the available pools
+	for _, pool := range mcpPoolsAll {
+		if pool.Name != ctrlcommon.MachineConfigPoolMaster && pool.Name != ctrlcommon.MachineConfigPoolWorker {
+			continue
+		}
+		managedKey := getManagedKeyDefaultContainerRuntime(pool)
+		mc, err := ctrl.client.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), managedKey, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return fmt.Errorf("error checking for %s machine config: %w", managedKey, err)
+		}
+		// continue to the next MC if this already exists
+		if mc != nil && !errors.IsNotFound(err) {
+			continue
+		}
+		// Check if ContainerRuntimeConfig exists
+		managedKeyForCtr, err := getManagedKeyCtrCfg(pool, ctrl.client, cfg)
```

Comment:
`getManagedKeyCtrCfg` returns the MC name of passed in cfg object. It can be one of existing MC or a new MC that does not exist.
There may be multiple containerruntimeconfig objects and only the last one takes effect.
I think for checking the current runtime configuration, we have to traverse the existing `99-pool-generated-containerruntimeconfig-x` machineconfigs in reverse order and check the first existing `DefaultRuntime` configuration. @yuqi-zhang  What do you think?

Reply:
Hmm, I guess this is a scenario where the design of the CRCC is a bit weird. Thinking through this scenario:
1. the user provides their own runtime pin, which generates into `99-pool-generated-containerruntimeconfig` and has contents for `/etc/crio/crio.conf.d/01-ctrcfg-defaultRuntime`
2. the user then adds a second config that sets e.g. pidLimit, which then translates to `/etc/crio/crio.conf.d/01-ctrcfg-pidsLimit` as `99-pool-generated-containerruntimeconfig-2`

Technically, the way we frame kubelet/containerruntimeconfigs, we don't merge them and we now 2 machineconfigs, but both containerruntimeconfigs still exist and are taking effect since they define different files. If I were to ever unpin via delete the runtime default, I guess it still works since `99-pool-generated-containerruntimeconfig` should get deleted alongside it?

Then in that case I guess we do have to parse through all the existing configuration... either through parsing all MCs that exist or all containerruntimeconfigs that exist (assuming that's the only way you can set a default runtime)

Reply:
> If I were to ever unpin via delete the runtime default, I guess it still works since 99-pool-generated-containerruntimeconfig should get deleted alongside it?

Yes, `99-pool-generated-containerruntimeconfig` will be deleted automatically when the corresponding ContainerRuntimeConfig objects are deleted.